### PR TITLE
primitive data types faq

### PR DESCRIPTION
**Pull request by davaya**
Probably will be moved into body section on JADN types (Primitive, Compound and Union)

### CODE REVIEW

**Comment by dlemire60** [comment]
BLUF: I like the direction but I'm not fond of the language.  This feels more like specification language than committee note language. The CN is intended as an explainer. I personally find the UML specifications nearly unreadable, so quoting them (e.g., lines 2499-2505) in an explainer doc doesn't seem "reader-friendly".

**Comment by dlemire60** [comment]
Wording is definitely improved. My read of the table in UML Spec section 21.2 indicates that the UnlimitedNatural number primitive includes the value zero, but your text and table here indicates one as the lowest value. Need to resolve the discrepancy.

File: imjadn-v1.0-cn02.md [hunk 1]
```diff
@@ -2488,28 +2488,38 @@ regarding JADN.
...
+The Universal Modeling Language [UML](#uml) says in section 21:
+"*The PrimitiveTypes package is an independent package that defines a set of
+reusable PrimitiveTypes that are commonly used in the definition of metamodels.*"
 
-###### Table D-1 -- UML and JADN Basic Type Equivalence
+JADN uses the same UML primitive types with two exceptions:
+* JADN defines a Binary (byte sequence) primitive type, UML does not.
+* UML defines an UnlimitedNatural primitive type, JADN does not.
+
+Natural numbers (integers greater than zero) are a subset of integers,
```
Comment (dlemire60) [inline imjadn-v1.0-cn02.md]: UML Spec in 21.2 says "An instance of UnlimitedNatural is a value in the (infinite) set of natural numbers (0, 1, 2…) plus
unlimited." So I believe line 2499 should say "(integers greater than _or equal to_ zero)".

File: imjadn-v1.0-cn02.md [hunk 2]
```diff
@@ -2488,28 +2488,38 @@ regarding JADN.
...
-|     Integer      |    Integer     |
+|       ---        |     Binary     |
 |     Boolean      |    Boolean     |
-|      String      |     String     |
-| UnlimitedNatural | Integer {0..*} |
+|     Integer      |    Integer     |
 |       Real       |     Number     |
-|      _xxx_       |     Binary     |
+|      String      |     String     |
+| UnlimitedNatural | Integer {1..*} |
```
Comment (dlemire60) [inline imjadn-v1.0-cn02.md]: As per comment above on line 2499, this should have 
```
|  UnlimitedNatural | Integer {0..*} |
```

**Review by dlemire60 — CHANGES_REQUESTED** [review]

File: imjadn-v1.0-cn02.md (hunk 1 above)
Comment (davaya) [reply imjadn-v1.0-cn02.md]: [Britannica](https://www.britannica.com/science/natural-number) says "natural number, any number in the set of positive integers{1, 2, 3…} and sometimes zero. The term natural likely refers to humanity’s innate understanding and use of positive discrete values. Sometimes called “counting numbers,” natural numbers predate recorded history, as it is believed that prehistoric humans had some sense of determining differences in quantities. Whether zero is considered a natural number differs across definitions, and there is no general consensus on its inclusion or exclusion from the set.

[Carnegie Mellon](http://mathfoundations.lti.cs.cmu.edu/class2/naturals.html) offers an interesting tidbit: "At least some of the controversy regarding 0 is due to the fact that the number 0 was a relatively late entrant to the pantheon of numbers, and is, in some sense, not natural (one elephant differs from one cat, but how do zero elephants differ from zero cats?).

[Many other sources](https://arbs.nzcer.org.nz/types-numbers), though none authoritative for the world of mathematics, declare whole numbers and natural numbers to be different.

But UML says what it says, so I'll make the changes.

Comment (dlemire60) [reply imjadn-v1.0-cn02.md]: I have a very good mathematician friend who’s an administrator / professor at Hopkins, author of several books on mathematics, if you want another opinion.  😊

But if we’re comparing JADN to UML I guess we at least need to represent UML as it is.

**Review by dlemire60 — APPROVED** [review]

**Pull request merged**